### PR TITLE
Add custom tags to machines

### PR DESCRIPTION
**Pull request by mcbenjemaa**
This PR add Custom Tags to VM while preserving the current tags.

### CODE REVIEW

File: internal/service/vmservice/vm.go [hunk 1]
```diff
@@ -225,6 +239,20 @@ func reconcileVirtualMachineConfig(ctx context.Context, machineScope *scope.Mach
...
+	// custom tags
+	if machineScope.ProxmoxMachine.Spec.Tags != nil {
+		tags := strings.Split(machineScope.VirtualMachine.VirtualMachineConfig.Tags, ";")
+		length := len(tags)
+		for _, tag := range machineScope.ProxmoxMachine.Spec.Tags {
+			if !machineScope.VirtualMachine.HasTag(tag) {
+				tags = append(tags, tag)
+			}
+		}
+		if len(tags) > length {
```
Comment (avorima) [inline internal/service/vmservice/vm.go:251]: this check does not account for removal of tags. is this intended?

File: internal/webhook/proxmoxmachine_webhook.go [hunk 2]
```diff
@@ -131,3 +144,14 @@ func validateNetworkDeviceMTU(device *infrav1.NetworkDevice) error {
 
 	return nil
 }
+
+func validateTags(tags []string) error {
+	re := regexp.MustCompile(`^[a-zA-Z0-9-_]*$`)
```
Comment (avorima) [inline internal/webhook/proxmoxmachine_webhook.go:149]: Should be `+` not `*` to prevent empty string. Also, have you tried `+kubebuilder:validation:Pattern`?

File: internal/service/vmservice/vm.go (hunk 1 above)
Comment (mcbenjemaa) [reply internal/service/vmservice/vm.go:251]: Yes, so we don't delete existing tags. 
The IP tag i mean.
Which means tags are also immutable as per the machine.

File: internal/webhook/proxmoxmachine_webhook.go (hunk 2 above)
Comment (mcbenjemaa) [reply internal/webhook/proxmoxmachine_webhook.go:149]: Hmmm, 

Let me check

File: internal/service/vmservice/vm.go (hunk 1 above)
Comment (avorima) [reply internal/service/vmservice/vm.go:251]: then you need to add this to the webhook or a CRD kubebuilder annotation

File: internal/service/vmservice/vm_test.go [hunk 3]
```diff
@@ -427,6 +430,27 @@ func TestReconcileVirtualMachineConfig_ApplyConfig(t *testing.T) {
 	require.EqualValues(t, task.UPID, *machineScope.ProxmoxMachine.Status.TaskRef)
 }
 
+func TestReconcileVirtualMachineConfigTags(t *testing.T) {
```
Comment (wikkyk) [inline internal/service/vmservice/vm_test.go:433]: Can you add a test case for multiple tags and for zero/empty tags?

**Comment by firecyberice** [comment]
closes #72
